backend/logic/courses/courses.js

```diff
@@ -29,6 +29,43 @@ var queryCourseByTag = async function queryCourseByTag(tag) {
   }
 };
 
+/**
+ * Adds a course into the database
+ * @author Mathieu Savoie
+ * @param {String} courseCode
+ * @param {String} title
+ * @param {String} departement
+ * @returns true if successful
+ * @throws error if MySQL connection failed
+ *         invalid format courses if JSON format is incorrect
+ *         false if insertion failed
+ */
+var addCourse = async (courseCode, title, departement, phasedOut) => {
+  
+  
+  // Verifying proper format
+  await format.verifyCourseCode(courseCode);
+  await format.verifyTitle(title);
+  await format.verifyDepartmentSubName(departement);
+  await format.verifyPhaseOut(phasedOut);
+  
+  // Connect to database
+  let connection = await mysql.getNewConnection();
+  
+   try {
+    await connection.beginTransaction();
```

Comment:
```suggestion
```
Unnecessary transaction since you are executing a single query. Use transactions when you have multiple queries that modify the db or depend on each other.